### ISSUE TITLE
Error compiling on Linux

### ISSUE THREAD

**Issue by einstein95** [nesbox/TIC-80#352]
Arch Linux, latest packages; compiling from latest Git.

```
src/ext/net/SDLnet.c: In function ‘SDLNet_ResolveHost’:
src/ext/net/SDLnet.c:152:45: error: ‘struct hostent’ has no member named ‘h_addr’
                 SDL_memcpy(&address->host,hp->h_addr,hp->h_length);
                                             ^~
src/ext/net/SDLnet.c: In function ‘SDLNet_GetLocalAddresses’:
src/ext/net/SDLnet.c:196:19: error: storage size of ‘conf’ isn’t known
     struct ifconf conf;
                   ^~~~
src/ext/net/SDLnet.c:207:21: error: ‘caddr_t’ undeclared (first use in this function); did you mean ‘__caddr_t’?
     conf.ifc_buf = (caddr_t) data;
                     ^~~~~~~
                     __caddr_t
src/ext/net/SDLnet.c:207:21: note: each undeclared identifier is reported only once for each function it appears in
src/ext/net/SDLnet.c:207:30: error: expected ‘;’ before ‘data’
     conf.ifc_buf = (caddr_t) data;
                              ^~~~
src/ext/net/SDLnet.c:215:16: error: dereferencing pointer to incomplete type ‘struct ifreq’
         if (ifr->ifr_addr.sa_family == AF_INET) {
                ^~
src/ext/net/SDLnet.c:196:19: warning: unused variable ‘conf’ [-Wunused-variable]
     struct ifconf conf;
                   ^~~~
make: *** [makefile:151: bin/SDLnet.o] Error 1
```

Only changes made is in the following patch:
```diff
diff --git a/makefile b/makefile
index 51baaa9..ba1f515 100644
--- a/makefile
+++ b/makefile
@@ -27,8 +27,7 @@ LINUX_INCLUDES= \
        `pkg-config --cflags gtk+-3.0`
 
 LINUX64_LIBS= \
-       `pkg-config --libs gtk+-3.0` \
-       -Llib/linux
+       `pkg-config --libs gtk+-3.0`
 
 LINUX32_LIBS= \
        `pkg-config --libs gtk+-3.0` \
@@ -145,7 +144,7 @@ bin/run.o: src/run.c $(TIC80_H) $(TIC_H)
        $(CC) $< $(OPT) $(INCLUDES) -c -o $@
 
 bin/file_dialog.o: src/ext/file_dialog.c $(TIC80_H) $(TIC_H)
-       $(CC) $< $(OPT) $(INCLUDES) -c -o $@
+       $(CC) $< $(OPT) $(INCLUDES) $(LINUX_INCLUDES) -c -o $@
 
 bin/md5.o: src/ext/md5.c $(TIC80_H) $(TIC_H)
        $(CC) $< $(OPT) $(INCLUDES) -c -o $@
@@ -250,10 +249,10 @@ bin/jsapi.o: src/jsapi.c $(TIC80_H)
        $(CC) $< $(OPT) $(INCLUDES) -c -o $@
 
 bin/luaapi.o: src/luaapi.c $(TIC80_H)
-       $(CC) $< $(OPT) $(INCLUDES) -c -o $@
+       $(CC) $< $(OPT) -fPIC $(INCLUDES) -c -o $@
 
 bin/duktape.o: src/ext/duktape/duktape.c $(TIC80_H)
-       $(CC) $< $(OPT) $(INCLUDES) -c -o $@
+       $(CC) $< $(OPT) -fPIC $(INCLUDES) -c -o $@
 
 TIC80_SRC = src/tic80.c src/tic.c src/ext/blip_buf.c src/jsapi.c src/luaapi.c src/ext/duktape/duktape.c
 TIC80_O = bin/tic80.o bin/tic.o bin/tools.o bin/blip_buf.o bin/jsapi.o bin/luaapi.o bin/duktape.o bin/gif.o
```
This is mainly due to the libraries supplied not being compiled with -fPIC, and thus throwing an error when compiled against on a different system.

**Comment by nesbox** [comment]
pls create a pull request to .48
https://github.com/nesbox/TIC-80/tree/tic_0.48.0
thanks

**Comment by josefnpat** [comment]
This patch doesn't seem to work without hunk failures. I will rebuild the patch, and test on Ubuntu and Arch linux, and PR it against 0.48 when I have time tonight.

**Comment by josefnpat** [comment]
@einstein95 I was unable to apply this patch and have it still compile on ubuntu 16.04. I will put some more research into this soon.

**Comment by nesbox** [comment]
duplicate of #367
